### ISSUE TITLE
tmk-query/tmk-query-with-faiss at scale

### ISSUE THREAD

**Issue by jcohenho** [facebook/ThreatExchange#227]
Is there any idea of what limitations are for a given haystack.txt file? How long would it take for tmk-query or tmk-query-with-faiss to run against a file with a hundred thousand or one million rows of tmk filenames? I'm assuming the processing is bound by GPU and available memory, but don't have any idea as to how long it would take or how much memory usage is involved. Maybe it's a good idea to split up the haystack.txt into multiple haystack.txt files and add some form of multi-threading/parallelization.  

Side note: Is there another forum to discuss new features and contributions to this project? I'd like to learn more but I don't want to spam the Issues section.

**Comment by johnkerl** [comment]
Hi @jcohenho !

I don't know about max sizes. We do need to well-integrate the with-faiss bit. Without that you're limited to maybe < 100K haystack size. With, I don't know -- FAISS itself scales into an absurdly large number of decimal places of volume but I don't know the cap once integrated into this TMK project.

And re forum, no I think this is the best one. Please don't worry about spamming. :D

**Comment by jcohenho** [comment]
Hi @johnkerl,  

I don't see the `.tmk-query-with-faiss` file. Was it included in the makefile? How can I test it out?

**Comment by johnkerl** [comment]
:) Yes, sorry about that ... on the one hand we want to be engaged with the community in every area but on the other hand TMK is one of many things we work on and sadly we haven't gotten the FAISS integration connected up. :(

**Comment by jcohenho** [comment]
@johnkerl No worries, appreciate your hard work! I can manage with tmk-query and use multiple haystack files for now :).

**Comment by jcohenho** [comment]
@johnkerl Hi again! I am unable to successfully complete a `tmk-query` with a haystack of 15,000 rows. I was able to get a successful run with 10,000, but I get a `killed` response when going higher. 

```
production_hashing_tools/tmk-query --c1 0.8 --c2 0.8 needles.txt haystack.txt
Killed
```

Any ideas? Is it possible to see an output while `tmk-query` is running? I tried adding the `--verbose` argument but it just gives more details in the final output for each .tmk file from the needles file.
